heft-plugins/heft-lint-plugin/src/LintPlugin.ts

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import { FileSystem } from '@rushstack/node-core-library';
+import type {
+  HeftConfiguration,
+  IHeftTaskSession,
+  IHeftTaskPlugin,
+  IHeftTaskRunHookOptions,
+  IScopedLogger
+} from '@rushstack/heft';
+import type { IChangedFilesHookOptions, ITypeScriptPluginAccessor } from '@rushstack/heft-typescript-plugin';
+
+import { Eslint } from './Eslint';
+import { Tslint } from './Tslint';
+import type { LinterBase } from './LinterBase';
+import type { IExtendedProgram, IExtendedSourceFile } from './internalTypings/TypeScriptInternals';
+
+const PLUGIN_NAME: string = 'LintPlugin';
+const ESLINTRC_JS_FILENAME: string = '.eslintrc.js';
+const ESLINTRC_CJS_FILENAME: string = '.eslintrc.cjs';
+
+export default class LintPlugin implements IHeftTaskPlugin {
+  private readonly _lintingPromises: Promise<LinterBase<unknown>[]>[] = [];
+
+  // These are initliazed by _initAsync
+  private _initPromise!: Promise<void>;
+  private _eslintToolPath: string | undefined;
+  private _eslintConfigFilePath: string | undefined;
+  private _tslintToolPath: string | undefined;
+  private _tslintConfigFilePath: string | undefined;
+
+  public apply(taskSession: IHeftTaskSession, heftConfiguration: HeftConfiguration): void {
+    // Use the changed files hook to kick off linting asynchronously
+    taskSession.requestAccessToPluginByName(
+      '@rushstack/heft-typescript-plugin',
+      'TypeScriptPlugin',
+      (accessor: ITypeScriptPluginAccessor) => {
+        // Hook into the changed files hook to kick off linting, which will be awaited in the run hook
+        accessor.onChangedFilesHook?.tap(PLUGIN_NAME, (changedFilesHookOptions: IChangedFilesHookOptions) => {
+          this._lintingPromises.push(
```

Comment:
What if you do:
```ts
const lintPromise: Promise<LinterBase<unknown>[]> = this._lintAsync(taskSession, heftConfiguration, changeFilesHookOptions.program, changedFilesHookOptions.changedFiles);
lintPromise.catch(() => { /* This error will be surfaced during run() */ });
this._lintingPromises.push(lintPromise);
```

Then don't bother swallowing the error inside of `_lintAsync`.